ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -87,9 +80,79 @@ int main()
     }
 }
 
-cv::Mat pointCloudToRGB(const Zivid::PointCloud &pointCloud)
+cv::Mat getImage3D(Zivid::Camera &camera, Zivid::Application &zivid)
+{
+    std::cout << "3D mode" << std::endl;
+
+    std::cout << "Configuring the camera settings" << std::endl;
+    camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+           << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+
+    std::cout << "Capturing a frame" << std::endl;
+    const auto frame = camera.capture();
+
+    std::cout << "Setting up visualization" << std::endl;
+    Zivid::CloudVisualizer vis;
+    zivid.setDefaultComputeDevice(vis.computeDevice());
+
+    std::cout << "Displaying the point cloud" << std::endl;
+    vis.showMaximized();
+    vis.show(frame);
+    vis.resetToFit();
+
+    std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+    vis.run();
+
+    std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+    const auto pointCloud = frame.getPointCloud();
+
+    return pointCloudToBGR(pointCloud);
+}
+
+cv::Mat getImage2D(Zivid::Camera &camera)
 {
-    cv::Mat rgb(static_cast<int>(pointCloud.height()),
+    std::cout << "2D mode" << std::endl;
+
+    std::cout << "Configuring the camera settings" << std::endl;
+    Zivid::Settings2D settings = Zivid::Settings2D();
+    settings.set(Zivid::Settings2D::ExposureTime{ std::chrono::microseconds{ 10000 } });
+    settings.set(Zivid::Settings2D::Gain{ 2.0 });
+    settings.set(Zivid::Settings2D::Iris{ 27 });
+    settings.set(Zivid::Settings2D::Brightness{ 1.0 });
+
+    std::cout << "Capturing a 2D frame" << std::endl;
+    const auto frame = camera.capture2D(settings);
+
+    std::cout << "Getting RGBA8 image from Frame2D" << std::endl;
+    const auto image = frame.image<Zivid::RGBA8>();
+
+    std::cout << "Converting RGBA8 image to OpenCV format" << std::endl;
+
+    return imageToBGR(image);
+}
+
+std::string getInput()
+{
+    std::string command;
+    std::getline(std::cin, command);
+    return command;
+}
+
+cv::Mat imageToBGR(const Zivid::Image<Zivid::RGBA8> &image)
+{
+    cv::Mat rgbaMat(static_cast<int>(image.height()),
```

Comment:
why cast around height and not width?  You can probably remove the static_cast unless the CI complains about this.

Reply:
I remember now.. I did it like this to check if the cast was necessary...